### PR TITLE
Ns/uoe 8382

### PR DESCRIPTION
**Pull request by pm-nitin-shirsat**
## Type of change

- [ ] Bugfix
- [x] Feature
- [ ] New bidder adapter   
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Does this change affect user-facing APIs or examples documented on http://prebid.org?
- [ ] Other

## Description of change

As a Developer, I want to change the translator request from POST to GET so that there is no delay in loading of payload and headers

- test parameters for validating bids
```
{
  bidder: '<bidder name>',
  params: {
    // ...
  }
}
```

Be sure to test the integration with your adserver using the [Hello World](/integrationExamples/gpt/hello_world.html) sample page.

- contact email of the adapter’s maintainer
- [ ] official adapter submission

For any changes that affect user-facing APIs or example code documented on http://prebid.org, please provide:

- A link to a PR on the docs repo at https://github.com/prebid/prebid.github.io/

### CODE REVIEW

File: modules/pubmaticBidAdapter.js [hunk 1]
```diff
@@ -1069,6 +1069,38 @@ export function prepareMetaObject(br, bid, seat) {
...
 }
 
+/**
+ * Checks url length is not exceeding the configured length.
+ * @param {*} payload - payload data
+ * @param {*} maxLength - length of URL characters
+ * @returns boolean
+ */
+function hasUrlLengthIsNotExceeding(payload, maxLength) {
+  const encodedPayload = btoa(JSON.stringify(payload));
```
Comment (pm-harshad-mane) [inline modules/pubmaticBidAdapter.js:1079]: why are we encoding the payload with base64? won't prebid encode the URL params?

**Comment by pm-harshad-mane** [comment]
There are no unit test cases added for the new changes. 
What is the plan for unit test cases?

**Comment by pm-nitin-shirsat** [comment]
> There are no unit test cases added for the new changes. What is the plan for unit test cases?

As this will be the experimental change so I haven't written Unit test cases. I will write it.

File: modules/pubmaticBidAdapter.js (hunk 1 above)
Comment (pm-nitin-shirsat) [reply modules/pubmaticBidAdapter.js:1079]: @pm-harshad-mane - Yes, prebid encodes the URL parameters. but we want to get the length of the URL before sending it to the prebid to call (which sends it to the server) so we can compare it with a provided maxLength.

**Comment by pm-nitin-shirsat** [comment]
Yes will work on the comments, and once completed will give it to review again. Thank you!

File: modules/pubmaticBidAdapter.js (hunk 1 above)
Comment (pm-harshad-mane) [reply modules/pubmaticBidAdapter.js]: comparing the maxLength is fine.
my question is about encoding the data in base64 using btoa even-though we know that the data will be URLencoded, why do we need to encode it with base64? can we avoid using `btoa` completely?

**Pull request merged**